typespec-tests/src/test/java/com/payload/multipart/MultipartTests.java

```diff
@@ -36,19 +43,60 @@ public void testBasic() {
 
     @Test
     public void testJson() {
-        client.jsonPart(new JsonPartRequest(new Address("X"), BinaryData.fromFile(FILE)));
+        client.jsonPart(new JsonPartRequest(
+                new Address("X"),
+                BinaryData.fromFile(FILE)));
     }
 
     @Test
     public void testJsonArray() {
-        client.jsonArrayParts(new JsonArrayPartsRequest(BinaryData.fromFile(FILE), Arrays.asList(new Address("Y"), new Address("Z"))).setProfileImageFilename("image.jpg"));
+        client.jsonArrayParts(new JsonArrayPartsRequest(
+                BinaryData.fromFile(FILE),
+                Arrays.asList(new Address("Y"), new Address("Z")))
+                .setProfileImageFilename("image.jpg"));
     }
 
     @Test
     public void testMultipleFiles() {
-        client.multiBinaryParts(new MultiBinaryPartsRequest(BinaryData.fromFile(FILE)).setPicture(BinaryData.fromFile(FileUtils.getPngFile())));
+        client.multiBinaryParts(new MultiBinaryPartsRequest(
+                BinaryData.fromFile(FILE))
+                .setPicture(BinaryData.fromFile(FileUtils.getPngFile())));
 
         // "picture" be optional
         asyncClient.multiBinaryParts(new MultiBinaryPartsRequest(BinaryData.fromFile(FILE))).block();
     }
+
+    @Test
+    public void testFileArray() {
+        // provide no filename
+        client.binaryArrayParts(new BinaryArrayPartsRequest(
+                "123",
+                Arrays.asList(BinaryData.fromFile(PNG_FILE), BinaryData.fromFile(PNG_FILE))));
+
+        // provide only 1 filename, when there are 2 files
+        // filename contains non-ASCII
+        asyncClient.binaryArrayParts(new BinaryArrayPartsRequest(
+                "123",
+                Arrays.asList(BinaryData.fromFile(PNG_FILE), BinaryData.fromFile(PNG_FILE)))
+                .setPicturesFilenames(Collections.singletonList("voilà.jpg"))).block();
+    }
+
+    @Test
+    public void testComplex() {
+        client.complex(new ComplexPartsRequest(
+                "123",
+                new Address("X"),
+                BinaryData.fromFile(FILE),
+                Arrays.asList(new Address("Y"), new Address("Z")),
+                Arrays.asList(BinaryData.fromFile(PNG_FILE), BinaryData.fromFile(PNG_FILE))));
+
+        // provide 3 filenames, when there are 2 files
+        asyncClient.complex(new ComplexPartsRequest(
+                "123",
+                new Address("X"),
+                BinaryData.fromFile(FILE),
+                Arrays.asList(new Address("Y"), new Address("Z")),
+                Arrays.asList(BinaryData.fromFile(PNG_FILE), BinaryData.fromFile(PNG_FILE)))
+                .setPicturesFilenames(Arrays.asList("picture1", "picture2", "picture3"))).block();
```

Comment:
cadl-ranch seems not assert for file name? do we need to verify the serialized result?

Reply:
Yeah. Currently I've verified the file name by human (i.e. me) reading the http log :-)

Verify automatically would require we do a more advanced verifier than https://github.com/Azure/autorest.java/blob/main/typespec-tests/src/test/java/com/parameters/bodyoptionality/BodyTests.java#L36-L44
Let me see if I want to do this...

Reply:
validation added